components/speedreader/rust/lib/src/readability/src/scorer.rs

```diff
@@ -744,7 +801,30 @@ pub fn clean<S: ::std::hash::BuildHasher>(
                 | local_name!("header")
                 | local_name!("footer")
                 | local_name!("embed")
+                | local_name!("textarea")
+                | local_name!("input")
+                | local_name!("select")
+                | local_name!("button")
                 | local_name!("aside") => true,
+                local_name!("h1") | local_name!("h2") => {
+                    // Delete remaining headings that may be duplicates of the title.
+                    let mut heading = String::new();
+                    dom::extract_text(&handle, &mut heading, true);
+                    if heading.len() == 0 {
+                        return true;
+                    }
+                    if title_tokens.len() == 0 {
+                        return false;
+                    }
+                    let heading_tokens = heading.split_whitespace().collect::<HashSet<_>>();
+                    let distance = title_tokens.difference(&heading_tokens).count() as f32;
+                    let similarity = 1.0 - distance / title_tokens.len() as f32;
+                    if similarity >= 0.75 {
+                        true
+                    } else {
+                        false
+                    }
```

Comment:
You don't need an `if`/`else` here, you can just return the condition directly

Reply:
fixed